src/commands/reactchannel.ts

```diff
@@ -0,0 +1,188 @@
+        'Sorry!  Could not find that emoji!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return { resultString: 'Could not find the emoji', result: {success: false}};
+    }
+
+    let channels: Collection<string, GuildChannel> = message.guild.channels;
+
+    if (!channels || !(channels.array()) || channels.array().length < 1) {
+      return this.couldNotFindTheChannel(message);
+    }
+
+    let textChannels: Collection<string, GuildChannel> = channels.filter((c) => c.type === 'text');
+
+    let channel: TextChannel = textChannels.find('id', channelIdOrName) as TextChannel;
+
+    if (!channel) {
+      channel = textChannels.find('name', channelIdOrName) as TextChannel;
+    }
+
+    if (!channel) {
+      return this.couldNotFindTheChannel(message);
+    }
+
+    let latestMessage = channel.lastMessageID;
+
+    if (!latestMessage) {
+      return this.couldNotFindMessageInChannel(message);
+    }
+
+    let targetMessage: Discord.Message;
+
+    // fetchMessage() throws if no message (e.g. last message deleted)
+    try {
+      targetMessage = await channel.fetchMessage(latestMessage);
+    } catch (ex) {
+      return this.couldNotFindMessageInChannel(message);
+    }
+
+    if (!targetMessage) {
+      return this.couldNotFindMessageInChannel(message);
+    }
+
+    const success = await this.reactService.tempReactToMessage(targetMessage, emoji, message);
+
+    if (success === false) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not apply the reaction!');
+      if (message.deletable) {
+        await message.delete(1000);
+      }
+      return {resultString: 'Could not apply the reaction', result: {success: false}};
+    }
+
+    return {};
+  }
+
+  async couldNotFindTheChannel(message: Discord.Message): Promise<ICommandResult> {
+    await this.errorMessageService.sendErrorMessage(message.channel,
+      'Sorry!  Could not find the given channel!');
+    if (message.deletable) {
+      await message.delete(1000);
+    }
+    return {resultString: 'Could not find the channel', result: {success: false}};
+  }
+
+  async couldNotFindMessageInChannel(message: Discord.Message): Promise<ICommandResult> {
+    await this.errorMessageService.sendErrorMessage(message.channel,
+      'Sorry!  Could not find a message in that channel!');
+    if (message.deletable) {
+      await message.delete(1000);
+    }
+    return {resultString: 'Could not find a message in that channel', result: {success: false}};
+  }
+
+  /**
+   * React to the message immediately above
+   * @param emojiName - The emoji name
+   * @param message - The command message
+   */
+  async reactNow(emojiName: string, message: Discord.Message): Promise<ICommandResult> {
+    const emoji = this.emojiService.getEmojiByName(emojiName);
+
+    if (!emoji) {
+      await this.errorMessageService.sendErrorMessage(message.channel,
+        'Sorry!  Could not find that emoji!');
```

Comment:
If this error message is super common (it looks like it is), maybe you should have a function in this command that just spits out this message.

Reply:
- [ ] If this error message is super common (it looks like it is), maybe you should have a function in this command that just spits out this message.